### DISCUSSION TITLE
Best server for Martin serving tiles from raster .mbtiles: RAM or Disk access speed ?

### DISCUSSION THREAD

**Discussion by nouryf** [maplibre/martin#959] [Q&A]
Hi,

I'm experimenting with Martin to serve mostly raster tiles from .mbtiles. I have a total of 5 .mbtiles. Each has a size of ~ 10 GB. I was wondering what would be a good server requirements to serve the tiles very fast ? Is there a recommended setup I could use to provision a server for this ?

I started with a server that has 4 GB RAM, 80 GB SSD disk size and 2 vCPU. The tiles are served by Martin just fine but I notice a little lag for their display: 2 to 3 seconds before they become crystal clear. The timing is obviously not just Martin. There the internet connexon speed and the front end app. I'm more focused on Martin for now and setting up a good server for speed performance.

Thanks

**Comment by nyurik** [comment]
I think Martin is mostly bound by disk/network IO (depending where your .mbitles resides). The 2-3s seems very high, and I doubt it is related to anything specific to Martin. If latency is so important, I think CDN will give you far better performance than anything else.

**Comment by nouryf** [comment]
Thank you @nyurik for your prompt answer. My server is in Europe (Germany) and I'm in the US. So from your answer that is probably the reason. CDN idea is great one. Can Martin serves tiles from an .mbtiles stored in a bucked with CDN enabled ? Looking at the [config doc](https://maplibre.org/martin/config-file.html), it appears that .mbtiles are accessed as local path on the same machine where Martin is running.

If accessing .mbtiles from a CDN enabled bucket (S3 or DigitalOcean for ex.) is possible assuming the Martin is installed and running on the same region (Germany for ex.), would that be a good configuration for latency ? 

Thanks

**Comment by nyurik** [comment]
CDN goes in front of Martin. Martin accesses the data directly - the closer the data to Martin, the better. There is some ideas on allowing Martin to access PMTiles from a bucket over HTTP - this could work well for cloud hosted scenarios.

**Reply by nouryf** [reply]
Thanks. So this is not possible yet with Martin unless Martin can access PMTiles over HTTP and not local access. Would this be a moot point since PMTiles are accessible through range requests using [PMTiles NodeJS](https://github.com/protomaps/PMTiles/blob/main/js/README.md) ? or the claim is that Martin would be a faster client ?
It would be interesting if MBTiles are supported through CDN with Martin. Unless they are already and I missed this in the documentation. If they are already, any doc outlining the setup ?
